singularity-eos/eos/eos_spiner_sie_transforms.hpp

```diff
@@ -0,0 +1,214 @@
+template<typename Data>
+struct DivideByCvTransform {
+  PORTABLE_INLINE_FUNCTION  DivideByCvTransform() = default;
+  
+  template <typename DataT_in>  
+  PORTABLE_INLINE_FUNCTION DivideByCvTransform(const DataT_in &data) : data_(data) {}
+
+  private: 
+    Data data_;
+    Real min_Cv_ = 1.0e-08;
+
+  public:
+    template <typename... Args>
+    PORTABLE_INLINE_FUNCTION
+    auto transform(Real e, Real rho, Args &&...) const {
+        const Real lRho = spiner_common::to_log(rho, data_.lRhoOffset);
+	const Real lE   = spiner_common::to_log(e, data_.lEOffset);
+        Real Cv = 1./data_.dTdE.interpToReal(lRho, lE);
+        Cv = std::max(Cv, min_Cv_);
+        return e / Cv;
+    }
+
+    template <typename... Args>
+    PORTABLE_INLINE_FUNCTION
+    auto inverse(Real e_transformed, Real rho, Real e_orig, Args &&...) const {
+        const Real lRho = spiner_common::to_log(rho, data_.lRhoOffset);
+	const Real lE   = spiner_common::to_log(e_orig, data_.lEOffset);
+	Real Cv = 1./data_.dTdE.interpToReal(lRho, lE);
+        Cv = std::max(Cv, min_Cv_);
+        return e_transformed * Cv;
+    }
+};
+
+
+
+//Divide by T^alpha
+template<typename Data>
+struct ScaleTransform {
+    PORTABLE_INLINE_FUNCTION  ScaleTransform() = default;
+    
+    template <typename DataT_in>
+PORTABLE_INLINE_FUNCTION ScaleTransform(const DataT_in &data) : data_(data) {}
+
+private:
+    Data data_;
+    Real min_T_ = 1.0e-08; //Good minimum temperature? 
+
+public: 
+    template <typename... Args>
+    PORTABLE_INLINE_FUNCTION
+    auto transform(Real e, Real rho, Args &&...) const {
+        Real lRho = spiner_common::to_log(rho, data_.lRhoOffset);
+        Real T = data_.T.interpToReal(lRho); //T?
+        T = std::max(T, min_T_);
+        return e / pow(T, 3);
+    }
+
+    template <typename... Args>
+    PORTABLE_INLINE_FUNCTION
+    auto inverse(Real e, Real rho, Args &&...) const {
+        Real lRho = spiner_common::to_log(rho, data_.lRhoOffset);
+        Real T = data_.T.interpToReal(lRho);
+        T = std::max(T, min_T_);
+        return e * pow(T, 3);
+    }
+};
+
+template<typename Data>
+struct AllTransform {
+PORTABLE_INLINE_FUNCTION  AllTransform() = default;
+   
+template <typename DataT_in>
+PORTABLE_INLINE_FUNCTION AllTransform(const DataT_in &data) : data_(data) {}
+
+
+
+private:
+    Data data_;
+    Real min_T_ = 1.0e-08; //Good minimum temperature? 
+    Real min_Cv_ = 1.0e-08;
+
+
+public:
+    template <typename... Args>
+    PORTABLE_INLINE_FUNCTION
+     auto transform(Real e, Real rho, Args &&...) const {
+
+     Real lRho = spiner_common::to_log(rho, data_.lRhoOffset);
+
+     Real e_cold = data_.sieCold.interpToReal(lRho);
+     Real T = data_.T.interpToReal(lRho);
+     T = std::max(T, min_T_);
+
+     Real e_transformed = e - e_cold;
+     Real lE   = spiner_common::to_log(e_transformed, data_.lEOffset);
+     Real Cv = 1./data_.dTdE.interpToReal(lRho, lE);
+     Cv = std::max(Cv, min_Cv_);
+     e_transformed = e_transformed / Cv;
```

Comment:
make sure to include the comments I made above about the double division for Cv.